Utils.cpp

```diff
@@ -0,0 +1,47 @@
+#include "Utils.h"
+#include <stringpool.h>
+#include <attribs.h>
+
+#include <string>
+
+using std::string;
+
+bool toObfuscate(bool flag, function* f, string attribute) {
+  /* skip declaration */
+  // return false;
+
+  /* skip external linkage */
+  // return false;
+
+  /* skip some functions that cannot currently be handled */
+  // return false;

```

Comment:
I guess these still have to be implemented before this gets merged?

Reply:
It can be implemented later.
The first two are modeled after ollvm, but GCC doesn't seem to need them. 
Now some functions cannot be processed, such as the function with switch. 
So far, I've only completed a part of it, and it's still to be tested.

Reply:
What is the problem with functions with a switch? I believe you just need to run a lowering pass first, but I thought I ran my pass after switch lowering?

Reply:
For example:
```
uint32_t target(uint32_t n) {
  uint32_t mod = n % 4;
  uint32_t result = 0;

  switch(mod){
    case 0:
      result = (n | 0xBAAAD0BF) * (2 ^ n);
      break;
    case 1:
      result = (n & 0xBAAAD0BF) * (3 + n);
      break;
    case 2:
      result = (n ^ 0xBAAAD0BF) * (4 | n);
      break;
    default:
      result = (n + 0xBAAAD0BF) * (5 & n);
  }

  return result;
}
```
`gcc -fPIC -fplugin=build/hellscape.so -fplugin-arg-hellscape-seed=deadbeef -fplugin-arg-hellscape-fla target.c -o target`
then：
```
*** WARNING *** there are active plugins, do not report this as a bug unless you can reproduce it without enabling any plugins.
Event                            | Plugins
PLUGIN_FINISH                    | hellscape
during RTL pass: expand
target.c: In function ‘target’:
target.c:9:3: internal compiler error: Segmentation fault
    9 |   switch(mod){
      |   ^~~~~~
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-9/README.Bugs> for instructions.
```

Reply:
GCC has a pass named lowerswitch. I tried to run it before fla, and it didn't work.

Reply:
Can we make an issue to track `switch` not working? Let's discuss there.

Reply:
OK.